### PR TITLE
DOC: Reference level name as Term of HDFStore.select query (#28791)

### LINKED ISSUE #28791: DOC: HDFStore.select documentation miss information about selecting MultiIndex
#### Code Sample

```python
import pandas as pd 
index = pd.MultiIndex.from_product([['a', 'b'], [1, 2]]) 
df = pd.DataFrame(range(4), index=index) 
store = pd.HDFStore('store.h5') 
store.append('dataframe', df) 
store.select('dataframe', where="level_0 in 'b' and level_1 in 2")         
                                                     
Out[3]: 
     0
b 2  3
```
#### Problem description

When using a MultiIndex in a HDFStore, you can select from the MultiIndex using the keyword level_0, level_1, …
This behaviour is not yet documented in the user guide.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : 02675e3376fea7d3cd484ace874d676c142cf003
python           : 3.7.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.3.1-arch1-1-ARCH
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : fr_FR.UTF-8
LOCALE           : fr_FR.UTF-8

pandas           : 0.26.0.dev0+446.g02675e337.dirty
numpy            : 1.16.3
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.0.3
setuptools       : 40.8.0
Cython           : 0.29.13
pytest           : 4.4.1
hypothesis       : 4.18.2
sphinx           : 1.8.5
blosc            : 1.8.1
feather          : None
xlsxwriter       : 1.1.7
lxml.etree       : 4.3.3
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: None
bs4              : 4.7.1
bottleneck       : 1.2.1
fastparquet      : 0.3.1
gcsfs            : None
lxml.etree       : 4.3.3
matplotlib       : 3.0.3
numexpr          : 2.6.9
odfpy            : None
openpyxl         : 2.6.2
pandas_gbq       : None
pyarrow          : 0.13.0
pytables         : None
s3fs             : 0.2.1
scipy            : 1.2.1
sqlalchemy       : 1.3.3
tables           : 3.5.1
xarray           : 0.12.1
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.1.7

</details>

### PR DESCRIPTION
**Pull request by nrebena**
Add documentation for default level name with HDFStore MultiIndex.

- [x] closes #28791
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -3933,6 +3934,29 @@ returned, this is equivalent to passing a
 
    store.select('df', "columns=['A', 'B']")
 
+Using level name when querying MultiIndex DataFrame.
```
Comment (jreback) [inline doc/source/user_guide/io.rst]: can you make this a sub-section (like using timedelta64), name it Query MultiIndex (also rename Using timedelta64 Query timedelta64)

File: doc/source/user_guide/io.rst [hunk 2]
```diff
@@ -3933,6 +3934,29 @@ returned, this is equivalent to passing a
...
+
+.. ipython:: python
+
+   store.select('df_mi', "foo=baz and bar=two")
+
+Using ``level_0`` and ``level_1`` for MultiIndex without name.
```
Comment (jreback) [inline doc/source/user_guide/io.rst]: give a statement about how the level names of ``level_i`` for the index of the level are available automatically if the level name is None

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
are there any other issues about this (can you do a search)

**Comment by nrebena** [comment]
> are there any other issues about this (can you do a search)

I did not find any other issue specifically about this (I did not even find it on StackOverflow). I guess everybody name its index…

I did find the issue #6208 that speficy that you cant use "index" as an level name (still true on master). I guess it is worth a note in section Storing MultiIndex DataFrames.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @nrebena

**Pull request merged**